skyportal/handlers/api/spatial_catalog.py

```diff
@@ -0,0 +1,379 @@
+    finally:
+        Session.remove()
+
+
+def delete_catalog(catalog_id):
+
+    log(f"Deleting catalog with ID {catalog_id}")
+
+    session = Session()
+    try:
+        catalog = session.scalar(
+            sa.select(SpatialCatalog).where(SpatialCatalog.id == int(catalog_id))
+        )
+        session.delete(catalog)
+        session.commit()
+
+        flow = Flow()
+        flow.push(
+            '*',
+            "skyportal/REFRESH_SPATIAL_CATALOGS",
+        )
+
+        log(f"Deleted catalog with ID {catalog_id}")
+    except Exception as e:
+        log(f"Unable to delete catalog: {e}")
+    finally:
+        Session.remove()
+
+
+class SpatialCatalogHandler(BaseHandler):
+    @auth_or_token
+    def post(self):
+        """
+        ---
+        description: Ingest a Spatial Catalog
+        tags:
+          - galaxies
+        requestBody:
+          content:
+            application/json:
+              schema: SpatialCatalogHandlerPost
+        responses:
+          200:
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: '#/components/schemas/Success'
+                    - type: object
+                      properties:
+                        data:
+                          type: object
+                          properties:
+                            id:
+                              type: integer
+                              description: New spatial catalog ID
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+
+        data = self.get_json()
+        catalog_name = data.get('catalog_name')
+        catalog_data = data.get('catalog_data')
+
+        if catalog_name is None:
+            return self.error("catalog_name is a required parameter.")
+        if catalog_data is None:
+            return self.error("catalog_data is a required parameter.")
+
+        # check RA bounds
+        if any([(x < 0) or (x >= 360) for x in catalog_data['ra']]):
+            return self.error("ra should span 0=<ra<360.")
+
+        # check Declination bounds
+        if any([(x > 90) or (x < -90) for x in catalog_data['dec']]):
+            return self.error("declination should span -90<dec<90.")
+
+        # check for cone or ellipse keys
+        if (not {'radius'}.issubset(set(catalog_data.keys()))) and (
+            not {'amaj', 'amin', 'phi'}.issubset(set(catalog_data.keys()))
+        ):
+            return self.error("error or amaj, amin, and phi required in field_data.")
+
+        with self.Session() as session:
+            stmt = SpatialCatalog.select(self.current_user).where(
+                SpatialCatalog.catalog_name == catalog_name
+            )
+            catalog = session.scalars(stmt).first()
+            if catalog is None:
+                catalog = SpatialCatalog(catalog_name=catalog_name)
+                session.add(catalog)
+                session.commit()
```

Comment:
and else? I think you want this to end here if there is already a catalog right? So consider using an `else:` statement + `self.error('...')`

Reply:
No then you are simply appending to a catalog.